code/datums/elements/strippable.dm

```diff
@@ -0,0 +1,531 @@
+
+/datum/element/strippable/proc/mouse_drop_onto(datum/source, atom/over, mob/user)
+	SIGNAL_HANDLER
+	if (user == source)
+		return
+
+	if (over == source)
+		return
+
+	var/mob/overmob = over
+	if (!istype(overmob))
+		return
+
+	if (!overmob.Adjacent(source))
+		return
+
+	if (!overmob.client)
+		return
+
+	if (overmob.client != user)
+		return
+
+	if (!isnull(should_strip_proc_path) && !call(source, should_strip_proc_path)(overmob))
+		return
+
+	var/datum/strip_menu/strip_menu
+
+	if (isnull(strip_menu))
+		strip_menu = new(source, src)
+		LAZYSET(strip_menus, source, strip_menu)
+
+	INVOKE_ASYNC(strip_menu, PROC_REF(tgui_interact), overmob)
+
+/// A representation of an item that can be stripped down
+/datum/strippable_item
+	/// The STRIPPABLE_ITEM_* key
+	var/key
+
+	/// Should we warn about dangerous clothing?
+	var/warn_dangerous_clothing = TRUE
+
+/// Gets the item from the given source.
+/datum/strippable_item/proc/get_item(atom/source)
+
+/// Tries to equip the item onto the given source.
+/// Returns TRUE/FALSE depending on if it is allowed.
+/// This should be used for checking if an item CAN be equipped.
+/// It should not perform the equipping itself.
+/datum/strippable_item/proc/try_equip(atom/source, obj/item/equipping, mob/user)
+	if ((equipping.flags_item & ITEM_ABSTRACT))
+		return FALSE
+	if ((equipping.flags_item & NODROP))
+		to_chat(user, SPAN_WARNING("You can't put [equipping] on [source], it's stuck to your hand!"))
+		return FALSE
+	if (ishuman(source))
+		var/mob/living/carbon/human/sourcehuman = source
+		if(HAS_TRAIT(sourcehuman, TRAIT_UNSTRIPPABLE) && !sourcehuman.is_mob_incapacitated())
+			to_chat(src, SPAN_DANGER("[sourcehuman] is too strong to force [equipping] onto them!"))
+			return
+	return TRUE
+
+/// Start the equipping process. This is the proc you should yield in.
+/// Returns TRUE/FALSE depending on if it is allowed.
+/datum/strippable_item/proc/start_equip(atom/source, obj/item/equipping, mob/user)
+	source.visible_message(
+		SPAN_NOTICE("[user] tries to put [equipping] on [source]."),
+		SPAN_NOTICE("[user] tries to put [equipping] on you.")
+	)
+
+	if (ismob(source))
+		var/mob/sourcemob = source
+		sourcemob.attack_log += text("\[[time_stamp()]\] <font color='orange'>[key_name(sourcemob)] is having [equipping] put on them by [key_name(user)]</font>")
+		user.attack_log += text("\[[time_stamp()]\] <font color='orange'>[key_name(user)] is putting [equipping] on [key_name(sourcemob)]</font>")
+
+	return TRUE
+
+/// The proc that places the item on the source. This should not yield.
+/datum/strippable_item/proc/finish_equip(atom/source, obj/item/equipping, mob/user)
+	SHOULD_NOT_SLEEP(TRUE)
+
+/// Tries to unequip the item from the given source.
+/// Returns TRUE/FALSE depending on if it is allowed.
+/// This should be used for checking if it CAN be unequipped.
+/// It should not perform the unequipping itself.
+/datum/strippable_item/proc/try_unequip(atom/source, mob/user)
```

Comment:
This proc is dead code; nothing uses it. And as such, everyone can multi-strip when it ought to only be MPs. I am fine with the action being queued though if you want to implement that though, but its not required.